keyboards/preonic/keymaps/brauner/keymap.c

```diff
@@ -279,6 +279,20 @@ static inline bool toggle_layer(enum preonic_layers layer, keyrecord_t *record)
     return false;
 }
 
+bool get_hold_on_other_key_press(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case MOD_TAP_LSFT_ESC:
+            /* Immediately select the hold action when another key is pressed. */
+            return true;
+        case MOD_TAP_LSFT_ENT:
+            /* Immediately select the hold action when another key is pressed. */
+            return true;
```

Comment:
Not a big difference, but writing it like this reduces repetition a bit, and would help adding other keycodes here easier and faster
```suggestion
        case MOD_TAP_LSFT_ESC:
        case MOD_TAP_LSFT_ENT:
            /* Immediately select the hold action when another key is pressed. */
            return true;
```

Reply:
I'm fine with that change. Note, that if you do allow that in the code you probably want to use `fallthrough` semantics going forward, i.e.,

```c
  #if HAVE_COMPILER_ATTR_FALLTHROUGH || __GNUC__ >= 7
  #define fallthrough __attribute__((__fallthrough__))
  #else
  #define fallthrough /* fall through */
  #endif

  bool get_hold_on_other_key_press(uint16_t keycode, keyrecord_t *record) {
      switch (keycode) {
          case MOD_TAP_LSFT_ESC:
              fallthrough;
          case MOD_TAP_LSFT_ENT:
              /* Immediately select the hold action when another key is pressed. */
              return true;
          default:
              /* Do not select the hold action when another key is pressed. */
              return false;
      }
  }
```

Then all places can be converted to use `fallthrough` and then you can ultimately enable `-Wimplicit-fallthrough=5`.

Reply:
updated